oci/toolchain.bzl

```diff
@@ -75,3 +77,82 @@ crane_toolchain = rule(
 For usage see https://docs.bazel.build/versions/main/toolchains.html#defining-toolchains.
 """,
 )
+
+
+RegistryInfo = provider(
+    doc = "Information about how to invoke the registry executable.",
+    fields = {
+        "launcher_path": "Path to the launcher script exporting a bash function named start_registry taking storage dir as the only argument.",
+        "registry_path": "Path to the registry executable for the target platform.",
+        "registry_files": """Files required in runfiles to make the registry executable available.
+
+May be empty if the registry_path points to a locally installed tool binary.""",
+    },
+)
+
+def _registry_toolchain_impl(ctx):
+    if ctx.attr.registry and ctx.attr.registry_path:
+        fail("Can only set one of registry or registry_path but both were set.")
+    if not ctx.attr.registry and not ctx.attr.registry_path:
+        fail("Must set one of registry or registry_path.")
```

Comment:
```suggestion
        fail("One of 'registry' or 'registry_path' attributes must be set.")
```

Reply:
we should fix this in the template as well. same as above.